client.js

```diff
@@ -121,27 +121,36 @@ RingpopClient.prototype._request = function _request(opts, endpoint, head, body,
         return;
     }
 
-    this.subChannel.waitForIdentified({
-        host: opts.host
-    }, function onIdentified(err) {
-        if (err) {
-            callback(err);
-            return;
-        }
+    try {
+        request();
+    } catch(e) {
+        callback(e);
+        return;
+    }
```

Comment:
@Raynos TChannel asserts on its input and raises exceptions. Is there API documentation that specifies the pre-conditions of the TChannel API (particularly, `waitForIdentified`, `request`) Otherwise, if there's not, you can't expect users to trace through TChannel code with every release to make sure they are being as defensive as they need to be to make sure that TChannel doesn't crash their application.

Some apps _do not_ continue on uncaught.

There are the assertions that we are being "bit" by:

```
574     assert(parts.length === 2, 'invalid destination');
575     var host = parts[0];
576     var port = parts[1];
577     assert(host !== '0.0.0.0', 'cannot connect to ephemeral peer');
578     assert(port !== '0', 'cannot connect to dynamic port');
```

It seems a